sdks/java/core/src/test/java/org/apache/beam/sdk/util/construction/UnboundedReadFromBoundedSourceTest.java

```diff
@@ -262,7 +261,7 @@ public void testInvokesSplitWithDefaultNumSplitsTooLarge() throws Exception {
     PipelineOptions options = PipelineOptionsFactory.create();
     List<?> splits = unboundedCountingSource.split(100, options);
     assertEquals(1, splits.size());
-    assertNotEquals(splits.get(0), unboundedCountingSource);
+    assertEquals(splits.get(0), unboundedCountingSource);
```

Comment:
Interesting, this means "assertNotEquals" didn't actually check that split has been invoked always